t/124-init-worker.t

```diff
@@ -756,3 +757,42 @@ GET /t
 ok
 --- no_error_log
 [error]
+
+
+
+=== TEST 20: disable init_worker_by_lua* in cache processes
+--- http_config
+    proxy_cache_path /tmp/cache levels=1:2 keys_zone=cache:1m;
+    init_worker_by_lua_block {
+        ngx.log(ngx.WARN, "running lua code in worker ", ngx.worker.id())
+    }
+--- config
+    location = /t {
+        return 200;
+    }
+--- request
+GET /t
+--- no_error_log
+[error]
+--- grep_error_log eval: qr/running lua code in worker [^,]+/
+--- grep_error_log_out
+running lua code in worker 0
+
+
+
+=== TEST 21: destory Lua VM in cache processes
+--- http_config
+    proxy_cache_path /tmp/cache levels=1:2 keys_zone=cache:1m;
+--- config
+    location = /t {
+        return 200;
+    }
+--- request
+GET /t
+--- error_log eval
+[
+qr/lua close the global Lua VM .+ in cache process/,
+qr/lua close the global Lua VM .+ in cache process/,
```

Comment:
I think we should just use a single regex here instead of an array since it's identical to all the repeated requests.

Reply:
Also, the use of `grep_error_log` would be better here since we can actually count the number of VM destructions (there should be two here, one for the cache loader and one for the cache manager).

Reply:
Additionally, we may want to add more tests for the cases when `init_worker_by_lua*` is used and when the privileged agent process is enabled?